gateway/api_definition.go

```diff
@@ -632,18 +622,18 @@ func (a APIDefinitionLoader) FromDir(dir string) []*APISpec {
 		}
 
 		def := a.ParseDefinition(f)
-		spec := a.MakeSpec(&def, nil)
-
-		_, _ = f.Seek(0, io.SeekStart)
-		_ = f.Close()
-
+		nestDef := nestedApiDefinition{APIDefinition: &def}
 		loader := openapi3.NewLoader()
 		oasDoc, err := loader.LoadFromFile(a.GetOASFilepath(path))
 		if err == nil {
-			spec.OAS.T = *oasDoc
-			_ = f.Close()
+			nestDef.OAS = &oas.OAS{T: *oasDoc}
 		}
 
+		spec := a.MakeSpec(nestDef, nil)
+
+		_, _ = f.Seek(0, io.SeekStart)
+		_ = f.Close()
```

Comment:
These functions return errors we should log if not nil. Would you create an func checkError(error, format string, args ...any) that matches "assert.NoError" signature without *T?

The alternative is to refactor functions to return wrapped errors, we follow that pattern elsewhere. Omitting error values and checks and logs is a red flag, because even if an error is unlikely or impossible to occur, it becomes impossible to surface the these errors without proper error logging or handling...

I can figure out a proposal in a separate PR so no immediate action needed here, but this is something we should address with consistency. I figure using a linter also works for this case, as we do not want to ignore a single error.